Code/Framework/AzCore/AzCore/Settings/CommandLine.cpp

```diff
@@ -103,142 +76,76 @@ namespace AZ
-                    // we support this too, for compatibility.
-
-                    currentArg = currentArg.substr(1);
-                    if (currentArg[0] == '-') // for -- extra
-                    {
-                        currentArg = currentArg.substr(1);
-                    }
-
-                    AZStd::size_t foundPos = AZ::StringFunc::Find(currentArg, "=");
-                    if (foundPos != AZStd::string::npos)
-                    {
-                        AZStd::string_view argumentView{ currentArg };
-                        AZStd::string_view option = AZ::StringFunc::StripEnds(argumentView.substr(0, foundPos));
-                        AZStd::string_view value = AZ::StringFunc::StripEnds(argumentView.substr(foundPos + 1));
-                        ParseOptionArgument(ToLower(option), value, nullptr);
-                        argumentParserState.m_currentOption.clear();
-                    }
-                    else
-                    {
-                        // its in this format -switchName switchvalue
-                        // (no equals)
-                        argumentParserState.m_currentOption = ToLower(currentArg);
-                        m_allValues.push_back({ argumentParserState.m_currentOption, "" });
-                    }
-                    return;
-                }
-            }
-
-            if (argumentParserState.m_currentOption.empty())
-            {
-                // Parse positional argument
-                m_allValues.push_back({ "", UnquoteArgument(currentArg) });
-            }
-            else
-            {
-                // Finish parsing of values for option argument
-                ParseOptionArgument(argumentParserState.m_currentOption, currentArg, &m_allValues.back());
-                argumentParserState.m_currentOption.clear();
+                m_allValues.push_back({ ToLower(newOption), optionValue });
             }
         }
     }
 
     void CommandLine::Parse(int argc, char** argv)
     {
         m_allValues.clear();
-
-        // Stores the state of arguments being parsed
-        // Allows the AddArgument function to update the state
-        ArgumentParserState parseState;
-
-        // Start on 1 because 0 is the executable name
-        for (int i = 1; i < argc; ++i)
+        if (argc == 0)
         {
-            if (argv[i])
-            {
-                AZStd::string_view currentArg = argv[i]; // this eats the / or -
-                AddArgument(currentArg, parseState);
-            }
+            return;
         }
+
+        Settings::CommandLineParserSettings parserSettings;
+        parserSettings.m_optionPrefixes = m_optionPrefixes;
+        parserSettings.m_parseCommandLineEntryFunc = [this](Settings::CommandLineArgument argument)
+        {
+            ParseArgument(argument.m_option, argument.m_value);
+            return true;
+        };
+
+        // Skip the zeroth argument for backwards compatibility
```

Comment:
This doesn't quite seem like backwards compatibility to me, it's just what we've defined this API as doing - it skips the executable name and moves on to the user-provided command-line settings.

```suggestion
        // Skip the zeroth argument since it is the executable name, not a user-provided command-line argument.
```

Reply:
Yes, I can update the comment. 

I consider the functionality of what it is doing as legacy because I don't see a good reason to skip the arg0 when parsing the command line, but it wasn't changed to keep from updating all the CommandLine Unit test check for positional parameters to increase by 1.